Sources/Modules/FilesModule.swift

```diff
@@ -1188,5 +1188,77 @@ public class FilesModule {
         }
     }
 
+    /// Creates a zip of multiple files and folders.
+    ///
+    /// - Parameters:
+    ///   - name: The name of the zip file to be created
+    ///   - items: Array of files or folders to be part of the created zip
+    ///   - completion: Returns a standard ZipDownload object or an error
+    public func createZip(
+        name: String,
+        items: [ZipDownloadItem],
+        completion: @escaping Callback<ZipDownload>
+    ) {
+        var body: [String: Any] = [:]
+        body["download_file_name"] = name
+        body["items"] = items.map { $0.bodyDictWithDefaultKeys }
+
+        boxClient.post(
+            url: URL.boxAPIEndpoint("/2.0/zip_downloads", configuration: boxClient.configuration),
+            json: body,
+            completion: ResponseHandler.default(wrapping: completion)
+        )
+    }
+
+    /// Creates a zip of multiple files and folders and downloads it.
+    ///
+    /// - Parameters:
+    ///   - name: The name of the zip file to be created
+    ///   - items: Array of files or folders to be part of the created zip
+    ///   - destinationURL: A URL for the location on device that we want to store the file once it has been downloaded
+    ///   - completion: Returns an empty response or an error
+    public func downloadZip(name: String, items: [ZipDownloadItem], destinationURL: URL, completion: @escaping Callback<ZipDownloadStatus>) {
+        createZip(name: name, items: items) { [weak self] result in
+            guard let self = self else {
+                completion(.failure(BoxSDKError(message: .instanceDeallocated("Unable to create Zip - FilesModule deallocated"))))
+                return
+            }
+
+            switch result {
+            case let .success(zip):
+                self.boxClient.download(
+                    url: URL.boxAPIEndpoint(zip.downloadUrl.absoluteString, configuration: self.boxClient.configuration),
```

Comment:
Is zip.downloadUrl a full endpoint like `https://dl.box.com/....`? If it is will URL.boxAPIEndpoint append `api.box.com` to the front?

Reply:
It is a full endpoint, so I thought the same would happen as well. But it downloads correctly with this setup, and instead when I try passing in `url: zip.downloadUrl` I get a 500 🤔  I'll take another look at it though, I might've missed something.